wp-includes/components/class-wp-service-worker-core-asset-caching-component.php

```diff
@@ -0,0 +1,92 @@
+		if ( is_admin() ) {
+			return;
+		}
+
+		$config = array(
+			'route'      => '^' . preg_quote( trailingslashit( includes_url() ), '/' ) . '.*',
+			'strategy'   => WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_FIRST,
+			'cache_name' => self::CACHE_NAME,
+			'expiration' => array(
+				'max_entries' => 14,
+			),
+		);
+
+		/**
+		 * Filters service worker caching configuration for core asset requests.
+		 *
+		 * @since 0.6
+		 *
+		 * @param array {
+		 *     Core asset caching configuration. If array filtered to be empty, then caching is disabled.
+		 *
+		 *     @type string     $route      Route. Regular expression pattern to match. See <https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-routing#.registerRoute>.
+		 *     @type string     $strategy   Strategy. Defaults to NetworkFirst.
+		 *                                  Even though assets should have far-future expiration, network-first is still preferred for development purposes.
+		 *                                  See <https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-strategies>.
+		 *     @type string     $cache_name Cache name. Defaults to 'uploaded-images'. This will get a site-specific prefix to prevent subdirectory multisite conflicts.
+		 *     @type array|null $expiration {
+		 *          Expiration plugin configuration. See <https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-expiration.ExpirationPlugin>.
+		 *
+		 *          @type int|null $max_entries     Max entries to cache. Defaults to 34.
+		 *                                          This limit the cached entries to the number of files loaded over network, e.g. JS, CSS, and PNG.
+		 *                                          The number 14 is derived from the 75th percentile of core assets used on pages served from
+		 *                                          WordPress sites, as indexed by HTTP Archive.
+		 *                                          See https://github.com/GoogleChromeLabs/pwa-wp/issues/265#issuecomment-706612536.
+		 *          @type int|null $max_age_seconds Max age seconds. Defaults to null.
+		 *     }
+		 *     @type array|null $broadcast_update   Broadcast update plugin configuration. Not included by default. See <https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-broadcast-update.BroadcastUpdatePlugin>.
+		 *     @type array|null $cacheable_response Cacheable response plugin configuration. Not included by default. See <https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-cacheable-response.CacheableResponsePlugin>.
+		 * }
+		 */
+		$config = apply_filters( 'wp_service_worker_core_asset_caching', $config );
+
+		if ( ! is_array( $config ) || ! isset( $config['route'], $config['strategy'] ) ) {
+			return;
+		}
+
+		$route = $config['route'];
+		unset( $config['route'] );
```

Comment:
See my above suggestion, things like this wouldn't be needed then.

Reply:
See my reply: https://github.com/GoogleChromeLabs/pwa-wp/pull/338/files#r523473404

Reply:
Left a reply, but either way, my previous comment on here would be irrelevant with my new suggestion there too.